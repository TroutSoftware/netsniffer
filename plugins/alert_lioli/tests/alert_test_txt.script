# Inspectors and spells are in place to attribute the correct flow
pcap $testdir/pcaps/google_http.pcap
cmp $testdir/alert_test_txt.expected.txt output.txt

-- cfg.lua --
output_to_file = { file_name = 'output.txt' }
output_to_stdout = { }

log_txt = { output = 'output_to_file' }

alert_lioli = { logger = 'log_txt',
                testmode = true }

stream = {}
stream_tcp = {}
stream_udp = {}
http_inspect = {}

wizard = {
    spells = { { service = 'http', proto = 'tcp', to_server = {'GET'}, to_client = {'HTTP/'} } }
}

binder = {
    { when = { service = 'http' }, use = { type = 'http_inspect' } },
    { use = { type = 'wizard' } }
}

ips = {
  include = 'lua.rules'
}

-- lua.rules --

alert ip any any -> any any (
  msg:"This is a log of an http header";

  http_header:field host;
  lioli_bind:host;
  content:"google";

  http_method;
  lioli_bind:method;
)
