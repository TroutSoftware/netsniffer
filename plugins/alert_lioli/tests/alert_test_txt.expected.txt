vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"10.67.21.59:48872209.85.202.100:80httpgoogle.comGET
-timestamp: 1970-01-01T00:00:00.000000000Z
-alert: "This is a log of an http header"
-principal: 10.67.21.59:48872
--addr: 10.67.21.59:48872
---ip: 10.67.21.59
---port: 48872
-endpoint: 209.85.202.100:80
--addr: 209.85.202.100:80
---ip: 209.85.202.100
---port: 80
-protocol: http
-host: google.com
-method: GET
^^^^^^^^^^^^^^^^^^^^^^^^
vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"10.67.21.59:48872209.85.202.100:80http
-timestamp: 1970-01-01T00:00:00.000000000Z
-log: "This is a log of an http header"
-principal: 10.67.21.59:48872
--addr: 10.67.21.59:48872
---ip: 10.67.21.59
---port: 48872
-endpoint: 209.85.202.100:80
--addr: 209.85.202.100:80
---ip: 209.85.202.100
---port: 80
-protocol: http
^^^^^^^^^^^^^^^^^^^^^^^^
vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"10.67.21.59:55904172.253.116.147:80httpwww.google.comGET
-timestamp: 1970-01-01T00:00:00.000000000Z
-alert: "This is a log of an http header"
-principal: 10.67.21.59:55904
--addr: 10.67.21.59:55904
---ip: 10.67.21.59
---port: 55904
-endpoint: 172.253.116.147:80
--addr: 172.253.116.147:80
---ip: 172.253.116.147
---port: 80
-protocol: http
-host: www.google.com
-method: GET
^^^^^^^^^^^^^^^^^^^^^^^^
vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"10.67.21.59:55904172.253.116.147:80http
-timestamp: 1970-01-01T00:00:00.000000000Z
-log: "This is a log of an http header"
-principal: 10.67.21.59:55904
--addr: 10.67.21.59:55904
---ip: 10.67.21.59
---port: 55904
-endpoint: 172.253.116.147:80
--addr: 172.253.116.147:80
---ip: 172.253.116.147
---port: 80
-protocol: http
^^^^^^^^^^^^^^^^^^^^^^^^
